tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/Serverless.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="Serverless.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation
+{
+    internal static class Serverless
+    {
+        private const string ExtensionFullPath = "/opt/extensions/datadog-agent";
+        private const string FunctionEnvame = "AWS_LAMBDA_FUNCTION_NAME";
+        private const string HandlerEnvName = "_HANDLER";
+        private const string IntegrationType = "Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.Lambda";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Serverless));
+
+        internal static void InitIfNeeded()
+        {
+            if (IsRunningInLambda())
+            {
+                Log.Debug("Sending CallTarget serverless integration definitions to native library.");
+                var serverlessDefinitions = GetServerlessDefinitions();
+                NativeMethods.InitializeProfiler(GetServerlessDefinitionsId(), serverlessDefinitions);
+                foreach (var def in serverlessDefinitions)
+                {
+                    def.Dispose();
+                }
+
+                Log.Information<int>("The profiler has been initialized with {count} serverless definitions.", serverlessDefinitions.Length);
+            }
+        }
+
+        internal static bool IsRunningInLambda()
+        {
+            return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(FunctionEnvame)) && File.Exists(ExtensionFullPath);
+        }
+
+        internal static NativeCallTargetDefinition[] GetServerlessDefinitions()
+        {
+            var serverlessDefinitions = Array.Empty<NativeCallTargetDefinition>();
+            try
+            {
+                LambdaHandler handler = new LambdaHandler(EnvironmentHelpers.GetEnvironmentVariable(HandlerEnvName));
+                string assymblyName = typeof(InstrumentationDefinitions).Assembly.FullName;
+                serverlessDefinitions = new NativeCallTargetDefinition[]
+                {
+                    new(handler.GetAssembly(), handler.GetFullType(), handler.GetMethodName(), handler.GetParamTypeArray(), 0, 0, 0, 65535, 65535, 65535, assymblyName, IntegrationType)
+                };
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+            }
+
+            return serverlessDefinitions;
+        }
+
+        internal static string GetServerlessDefinitionsId()
+        {
+            return "4D16D270B8DC62693B5681458364AF42";
```

Comment:
this is a random hex string

Reply:
just to be similar to others definitions ids, consider to use a random Guid :)

Reply:
✅